src/api/app/mixins/get_flags.rb

```diff
@@ -0,0 +1,34 @@
+module GetFlags
+  # FIXME: All of this would not be needed if creating an Package/Project would create all
+  #        the build, publish, debuginfo and useforbuild Flags. The only reason
+  #        this happens here is because rendering the XML does ignore the defaults.
+  # Returns a hash of arrays, sorted by repository.
+  # The arrays contain Flag objects of type, sorted by architecture.
+  # Like:
+  # {all: [Flag, Flag-i586, Flag-x86_64], 13.2: [Flag, Flag-i585, Flag-x86_64]}
+  def get_flags(flag_type)
+    the_flags = {}
+
+    # FIXME: A helper called in a model...
+    return the_flags unless FlagHelper::TYPES.has_key?(flag_type.to_s)
```

Comment:
Wouldn't it then be better to let it fail miserably so we are getting aware of such bugs early?